R/copy-model-from.R

```diff
@@ -180,32 +179,30 @@ copy_nonmem_model_from <- function(
 
 #' Copy a NONMEM control stream file
 #'
-#' Helper function to copy a NONMEM control stream file and optionally update the description, etc. in the new file.
+#' Helper function to copy a NONMEM control stream file and optionally update the new file.
 #' Note that any file existing at `.new_model_path` will be overwritten.
 #' @param .parent_model_path Path to the control stream to copy
 #' @param .new_model_path Path to copy the new control stream to
 #' @param .overwrite If `TRUE`, overwrite existing file at `.new_model_path`. If `FALSE` and file exists at `.new_model_path` error.
 #' @param .update_model_file If `TRUE`, the default, update the `$PROBLEM` line in the new control stream. If `FALSE`, `{.new_model}.[mod|ctl]` will be an exact copy of its parent control stream.
-#' @param .description Description of new model run. This will be passed into the `$PROBLEM` of the new control stream (if `.update_model_file=TRUE`).
 #' @keywords internal
-copy_control_stream <- function(.parent_model_path, .new_model_path, .overwrite, .update_model_file = FALSE, .description = NULL) {
+copy_control_stream <- function(.parent_model_path, .new_model_path, .overwrite, .update_model_file = FALSE) {
 
   if (fs::file_exists(.new_model_path) && !isTRUE(.overwrite)) {
     stop(glue("File already exists at {.new_model_path} -- cannot copy new control stream. Either delete old file or use `new_model({yaml_ext(.new_model_path)})`"))
   }
 
   if (.update_model_file) {
-    if (is.null(.description)) {
-      stop("If `.update_model_file` is TRUE, user must specify a `.description` for the new model.")
-    }
 
     # read parent control stream
     mod_str <- .parent_model_path %>% read_file()
 
     # replace the $PROBLEM line(s)
+    new_mod_id <- get_model_id(.new_model_path)
     mod_str <- str_replace(mod_str,
                            "\\$PROB(.|\n)*?\\$",
-                           as.character(glue("$PROBLEM {.description}\n\n$")))
+                           as.character(glue("$PROBLEM See {new_mod_id}.yaml. Created by rbabylon.\n\n$")))
```

Comment:
Not a huge deal but ... "Created by rbabylon." might suggest that rbabylon wrote out the control stream.  Not sure.  But is there a different wording that would only suggest that rbabylon only inserted the text in `$PROBLEM`?

```
$PROBLEM From rbabylon: see {new_mod_id}.yaml for details
```